aptos-move/framework/aptos-stdlib/sources/cryptography/curves.move

```diff
@@ -0,0 +1,217 @@
+module aptos_std::curves {
+    use aptos_std::type_info::type_of;
+
+    // Structs and consts.
+
+    // Fake structs representing group type.
+    struct BLS12_381_G1 {}
```

Comment:
Having these is great! We will also need some helper methods. Here's a few I can think of:

 - `is_prime_order<T>() -> bool` returns true if the group is of prime order
 - `get_group_order<T>() -> vector<u8>` returns the little-endian encoding (say) of the order of the group

Other functions might tell us about various features the implementation supports: e.g., hashing to the group, fast subgroup checks, etc. But we'll add these on a needed basis.

Reply:
when will a developer call `is_prime_order` or `get_group_order`?
I feel documentation + meaningful runtime error is good enough.

Reply:
Hm, I can't think of a use-case for `is_prime_order` but can't hurt to add it: it's just a `return true` or `return false` function.

`get_group_order` will be necessary when trying to figure out how wide a field element is and how many bits of information it can encode.

For example, an application might want to pack a 1024-bit integer into several field elements and needs to know how many field elements it needs. `get_floor_log_2_group_order` would suffice here.

There might be other settings where knowing the order is important. I vaguely recall examples of cryptosystems that used pairing-friendly groups of order $q$ and together with lattices of vectors with elements from $\mathbb{F}_q$. In order to create the lattice, one needs to know $q$.

I can't find that example though.

Reply:
The problem is we'd have to return the order as a little-endian or big-endian vector, since there's no support for big number arithmetic in Move yet... Maybe we can settle for `get_floor_log_2_group_order` for now?

Reply:
I just added `is_prime_order` and `group_order` (l-endian). User can derive the bit length/byte length easily?